src/java.desktop/share/classes/java/awt/Robot.java

```diff
@@ -383,8 +395,25 @@ private static void checkKeycodeArgument(int keycode) {
 
     /**
      * Returns the color of a pixel at the given screen coordinates.
+     *
+     * @implNote On Linux systems using Wayland, permission may be requested
+     * from a user to capture the screens via system dialog.
+     * When selected in this dialog, the permission to capture a set of screens
+     * can be saved for later reuse.<br>
+     * Use the {@link #revokeScreenCapturePermission()} to revoke the stored
+     * permission.
+     * <br>
+     * The pixel color may be black for screens for which no capture permission
+     * has been granted.
+     * <br>
+     * It is not recommended to call this method on EDT, as it may block the UI
+     * update until the user confirms his choice in the system dialog.
+     *
      * @param   x       X position of pixel
      * @param   y       Y position of pixel
+     * @throws  SecurityException if {@code readDisplayPixels} permission
+     *          is not granted, or user has denied screen capture for all
+     *          screens
```

Comment:
> Is there some subtlety about "ALL" screens ?

It is more like "the user has not allowed any of his screens to be captured."

In fact, we may not throw out this exception and just provide black pixels(and call it  "unspecified" behavior).

> Do you really mean all screens which are in the area being captured ?

> I added the word REQUIRED here, since if we only need screen 1 contents, surely it
doesn't matter if we don't have screen 2 permission ?

That's not exactly true.

There are screens that the user may or may not have given permission to capture. 
This permission(specifically `restore_token`) is one at a single moment in time, not several separate permissions for each screen. This restore_token is associated with the permissions for specific screens.

There is an area that the user wants to capture.

The intersection of the capture area and the screens allowed for capture will have actual data from the screen, otherwise - black pixels.

So if a user allows access only to screen A and then asks for pixels from screen B he will get black pixels.

This is about the same when we have a single monitor and we capture an area outside of it (e.g. -100, -100, 50, 50) - we get a black square.

> Does that also imply you can get a partial capture if it spans 2 screens and you only have permission for one of them ?

Yes, the plan is to provide what we can and leave the other parts black.

> We do also need to think forward to a pure wayland scenario - any differences there. ?

As a screen capture - should work as it is.

The only concern is the impossibility to specify and get the window coordinates.
This way we won't know where the window we are displaying is located.
It will affect the testing approach and we will have to add a method like `captureWindowContents` to check the window contents.